AssociateContactAccountHelper.cls

```diff
@@ -0,0 +1,43 @@
+/***********************************************************************************
+ * Class : AssociateContactAccountHelper
+ * Created By : Pranali Mahajan
+ -----------------------------------------------------------------------------------
+ * Description : AssociateContactAccountHelper class
+ -----------------------------------------------------------------------------------
+ * Version History:
+ * Version    Developer Name    Date          Detail Features
+ * 1.0        Pranali Mahajan                Intial Development
+ **********************************************************************************
+ ABC Containers require the ability to automatically associate a Contact created in their Salesforce Instance 
+ with the respective Account based on the email domain specified in the primary email address of the Contact. 
+ The association should happen real time as soon as a Contact record is created within the system. */
+
+public with sharing class AssociateContactAccountHelper {
+	public void associateContactToAccount(List<Contact> con){
+		List<String> contactEmaildomains = new List<String>();
+		
+        for(Contact contactlist:con)
```

Comment:
`contactlist` is not proper variable name as it's only holding single contact record.